### PR TITLE
[WM-1438] - Hackathon submission details kp (error messages included)

### PR DESCRIPTION
**Pull request by kpierre13**

### CODE REVIEW

File: src/pages/SubmissionDetails.js [hunk 1]
```diff
@@ -76,11 +84,17 @@ export const SubmissionDetails = ({ submissionId }) => {
...
-                size: { basis: 200, grow: 0 },
+                size: { basis: 220, grow: 0 },
                 field: 'state',
                 headerRenderer: () => h(Sortable, { sort, field: 'state', onSort: setSort }, ['Status']),
                 cellRenderer: ({ rowIndex }) => {
-                  return h(TextCell, [paginatedPreviousRuns[rowIndex].state])
+                  const failureStates = ['SYSTEM_ERROR', 'EXECUTOR_ERROR']
+                  if (failureStates.includes(paginatedPreviousRuns[rowIndex].state)) {
+                    return div({ style: { width: '100%', textAlign: 'center' } }, [
+                      div({ style: { marginBottom: '1rem', fontWeight: 'bold' } }, [h(TextCell, {}, [icon('warning-standard', { size: 18, color: 'red' }),['   Failed with error']])]),
```
Comment (salonishah11) [inline src/pages/SubmissionDetails.js:94]: Are the whitespaces in front of `Failed with error` text intended?

File: src/pages/SubmissionDetails.test.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 })
+  })
+
+  beforeEach(() => {
+    const getRunsMethod = jest.fn(() => Promise.resolve(runsData))
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runs: {
+            get: getRunsMethod
```
Comment (salonishah11) [inline src/pages/SubmissionDetails.test.js:58]: Should this now take in Run Set ID as an argument?

File: src/pages/SubmissionDetails.test.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    within(headers[4]).getByText('Data')
+    within(headers[5]).getByText('Logs')
+
+    // check data rows are rendered as expected
+    const cellsFromDataRow1 = within(rows[1]).queryAllByRole('cell')
+    expect(cellsFromDataRow1.length).toBe(6)
+    within(cellsFromDataRow1[0]).getByText('b7234aae-6f43-405e-bb3a-71f924e09825')
+    within(cellsFromDataRow1[1]).getByText('Failed')
+    within(cellsFromDataRow1[2]).getByText(/Jul 14, 2022/)
+    // TODO: Add 'within for 'duration' once WM-1461 is complete
```
Comment (salonishah11) [inline src/pages/SubmissionDetails.test.js:104]: I think the check for duration can be added now as it only relies on mock data and not the actual API response

File: src/pages/SubmissionDetails.test.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    within(cellsFromDataRow1[4]).getByText('View inputs')
+    // Oct 18, 2022, 10:44 AM
+    // 22:27:15.591Z
```
Comment (salonishah11) [inline src/pages/SubmissionDetails.test.js:107]: Remove these comments?

File: src/pages/SubmissionDetails.test.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      await fireEvent.click(within(headers[2]).getByRole('button'))
+    })
+
+    // Assert - rows are now sorted by submission timestamp in ascending order
+    const cellsFromUpdatedDataRow1 = within(rows[1]).queryAllByRole('cell')
+    expect(cellsFromUpdatedDataRow1.length).toBe(6)
+    within(cellsFromUpdatedDataRow1[0]).getByText('ea001565-1cd6-4e43-b446-932ac1918081')
+    within(cellsFromUpdatedDataRow1[1]).getByText('Submitted')
+    within(cellsFromUpdatedDataRow1[2]).getByText(/Jan 27, 2022, 5:27 PM/)
+    // TODO: Add 'within for 'duration' once WM-1461 is complete
```
Comment (salonishah11) [inline src/pages/SubmissionDetails.test.js:142]: Same comment about duration check here as well

File: src/pages/SubmissionDetails.test.js (hunk 2 above)
Comment (kpierre13) [reply src/pages/SubmissionDetails.test.js:58]: I think this was the question I was trying to ask yesterday but didn't know how to frame it

Comment (kpierre13) [reply src/pages/SubmissionDetails.test.js:58]: I think it should to reflect how we get the runs?

File: src/pages/SubmissionDetails.js (hunk 1 above)
Comment (kpierre13) [reply src/pages/SubmissionDetails.js:94]: They are

File: src/pages/SubmissionDetails.test.js (hunk 2 above)
Comment (salonishah11) [reply src/pages/SubmissionDetails.test.js:60]: Yes it should be same as the actual API being used in the components.

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**